Builderror with gdkpixbuf 2.0.26

Hey there. While packaging for nixpkgs I encountered this build error, which I assume is related to our gdkpixbuf bump.

```
Building library for gtk-sni-tray-0.1.6.0..
[1 of 3] Compiling Paths_gtk_sni_tray ( dist/build/autogen/Paths_gtk_sni_tray.hs, dist/build/Paths_gtk_sni_tray.o, dist/build/Paths_gtk_sni_tray.dyn_o )
[2 of 3] Compiling StatusNotifier.TransparentWindow ( src/StatusNotifier/TransparentWindow.hs, dist/build/StatusNotifier/TransparentWindow.o, dist/build/StatusNotifier/TransparentWindow.dyn_o )
[3 of 3] Compiling StatusNotifier.Tray ( src/StatusNotifier/Tray.hs, dist/build/StatusNotifier/Tray.o, dist/build/StatusNotifier/Tray.dyn_o )

src/StatusNotifier/Tray.hs:125:5: error:
    • Couldn't match type ‘Maybe Pixbuf’ with ‘Pixbuf’
      Expected type: IO (Maybe Pixbuf)
        Actual type: IO (Maybe (Maybe Pixbuf))
    • In a stmt of a 'do' block:
        sequenceA $ pixbufNewFromFile <$> maybeFile
      In the expression:
        do trayLogger DEBUG
             $ printf "Trying to load icon %s as filepath" name
           let nameString = T.unpack name
           fileExists <- doesFileExist nameString
           maybeFile <- if fileExists then
                            return $ Just nameString
                        else
                            fmap join
                              $ sequenceA $ getIconPathFromThemePath nameString <$> themePath
           ....
      In a stmt of a 'do' block:
        if hasIcon || hasPanelIcon then
            do let targetName = ...
               trayLogger DEBUG $ printf "Found icon %s in theme" name
               iconThemeLoadIcon themeForIcon targetName size themeLoadFlags
        else
            do trayLogger DEBUG
                 $ printf "Trying to load icon %s as filepath" name
               let nameString = ...
               fileExists <- doesFileExist nameString
               ....
    |
125 |     sequenceA $ pixbufNewFromFile <$> maybeFile
    |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

--- a/src/StatusNotifier/Tray.hs
+++ b/src/StatusNotifier/Tray.hs
@@ -1,5 +1,6 @@
 {-# LANGUAGE OverloadedStrings #-}
 {-# LANGUAGE OverloadedLabels #-}
+{-# LANGUAGE CPP #-}
 module StatusNotifier.Tray where
 
 import           Control.Concurrent.MVar as MV
@@ -123,7 +124,12 @@ getIconPixbufByName size name themePath = do
     maybeFile <- if fileExists
     then return $ Just nameString
     else fmap join $ sequenceA $ getIconPathFromThemePath nameString <$> themePath
-    sequenceA $ pixbufNewFromFile <$> maybeFile
+#if MIN_VERSION_gi_gdkpixbuf(2,0,26)
+    let handleResult = fmap join . sequenceA
+#else
+    let handleResult = sequenceA
+#endif
+    handleResult $ pixbufNewFromFile <$> maybeFile
 
 getIconPathFromThemePath :: String -> String -> IO (Maybe String)
 getIconPathFromThemePath name themePath = if name == "" then return Nothing else do
@@ -257,8 +263,9 @@ buildTray TrayParams { trayHost = Host
                           in return (thisTime, thisTime /= previous)
 
                         trayLogger DEBUG $
-                                   printf "Allocating image size %s, width %s, \
-                                           \ height %s, resize %s"
+                                   printf
+                                   ("Allocating image size %s, width %s," <>
+                                    " height %s, resize %s")
                                    (show size)
                                    (show actualWidth)
                                    (show actualHeight)

